### ISSUE TITLE
Some (all?) manual pages of the online manual have an extremely high memory footprint because they're not lazily loaded/unloaded

### ISSUE THREAD

**Issue by ahydronous** [NixOS/nixos-homepage#1521]
**Describe the glitch**
Pages like https://nixos.org/manual/nixos/unstable/options use 1.5-2GB (!) of memory when fully loaded.
This happens even on first load, without any page interaction.

**Screenshots**
![image](https://github.com/user-attachments/assets/80062d14-1305-4082-9f0d-9e0c12ec8e25)

**Additional context**
This terribly bogs down any memory-constrained device.

**Comment by thilobillerbeck** [comment]
Pinging @fricklerhandwerk here since splitting the manual is something that belongs to the docs team to my knowledge

**Comment by fricklerhandwerk** [comment]
Yes, it's high in my priority list and unless someone beats me to it I'll get this addressed eventually.

**Comment by JuryA** [comment]
It’s been almost a year since this issue was opened, and still no progress has been made — the entire current manual on www.nixos.org/manual is practically unusable!

This isn’t because I lack memory; I have 64 GB of RAM, with 50% readily available. The problem is that most mainstream browsers simply can’t handle it. As for trying to open it on a mobile device (whether an iPhone 16 Pro Max or an iPhone 12 Pro Max) — I don’t even bother anymore.

So who is this manual actually intended for? It’s clearly not for the “average user,” nor for those of us with ample system resources but no compatible browser.

I suppose it might be accessible in Lynx, but I’ll refrain from commenting further — I’d probably get too rude.

This is a similar issue (albeit in a slightly different form) as the one plaguing the Kubernetes API Reference Manual — also a single-page document.

What exactly is so brilliant about this single-page solution that I’m somehow missing, and why is everyone so enamored with it?

**Comment by JuryA** [comment]
**Proposed Solution**

1. **Default to Multi-page, Single-page as Optional**  
   - **Default Mode:** Split the manual into separate chapters or sections, minimizing the size of each page and accelerating load times.  
   - **Optional Mode:** Allow users who require a consolidated document (e.g., for offline reading or global search) to switch to the single-page view via a URL parameter or user preferences setting.

2. **Enhanced Fuzzy Search**  
   - Integrate advanced client-side full-text search (fuzzy matching) in both multi-page and single-page modes.  
   - This enables users to quickly locate content across chapters without relying on exact keywords, preserving the main advantage of a single document—easy searchability—while keeping multi-page mode user-friendly.

3. **Robust URL Handling and Redirects**  
   - Ensure all existing links to individual chapters or the full manual remain functional.  
   - Implement a simple redirect map (server-side rewrites or client-side redirects) to automatically route legacy paths to the new format (multi-page with anchors or single-page).

**Benefits**

- **Performance & Stability:** Dividing the manual into smaller pages significantly reduces memory usage and enhances compatibility.  
- **Usability & Flexibility:** Users who need a unified document retain the option of single-page mode.  
- **Searchability:** Built-in fuzzy search provides a fully-featured alternative to full-text search in a single-page document.  
- **Backward Compatibility:** Redirects guarantee that all bookmarks and external links continue to function without modification.

This proposal supports parallel workflows for the manual and community-driven, incremental implementation without disrupting existing links.

**Comment by ahydronous** [comment]
@JuryA that does not deal with the root cause and will probably make the same issue return further down the line.

Why is a *text only markup page* consuming up to **2 gigabytes of memory**?

**Comment by JuryA** [comment]
> @JuryA that does not deal with the root cause and will probably make the same issue return further down the line.
> 
> Why is a *text only markup page* consuming up to **2 gigabytes of memory**?

Couldn’t agree more—this endless “research” (a.k.a. root-cause spelunking instead of shipping a fix) has become a never-ending archaeological dig. Meanwhile, the crew luxuriates in its own little bubble, brandishing exotic browsers or resorting to nothing but curl.

**It’s infuriating that this single-page manual still won’t load in any conventional browser—even on a machine with 64 GB of RAM.**

I’d leap at the chance to help, but as a Nix novice, the manual is my lifeline—and each time I crack it open, my poor computer throws a tantrum and freezes solid. During those delightful freeze-ups, I half-mockingly channel Bob Widlar, fantasizing about “Widlarizing” the whole circus with a gargantuan hammer. Then I resign myself to another three-month Nix detox.

Frankly, I couldn’t care less about perfecting root-cause analysis—it’s a tertiary concern. I just need this damned system to work before I retire—very soon—and trade bug hunts for hammock naps. It’s infuriating, even heartbreaking, to be stranded in limbo while all those bright young Einsteins parade around. If the community expects contributors, let’s make contributing feasible—stop rabbit-holing and start delivering.
